rpmlint/checks/FileDigestCheck.py

```diff
@@ -198,11 +210,30 @@ def _matches_pkg(self, digest_group, pkg):
 
         return False
 
+    def _is_path_covered_by_digests(self, path):
+        """
+        Return true if there is a digest configuration that covers
+        provided file at given path.
+        """
+        parts = list(path.parts) + [None]
+        node = self.digest_configuration_locations
+        for part in parts:
+            if None in node:
+                return True
+            if part not in node:
+                return False
+            node = node[part]
+        return True
```

Comment:
I don't think this final return is correct. Consider we have a restriction for "/etc/cron.daily" and a package ships the file or dir "/etc". In that case the function would return `True`, although that is not really correct.

Reply:
also just from a security POV is would like to see the logic to return False unless something is found, which makes later modifications more reliable

Reply:
Sure.